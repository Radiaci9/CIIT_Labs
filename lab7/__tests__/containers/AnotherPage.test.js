import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import AnotherPage from '../../src/containers/AnotherPage'
import configureStore from 'redux-mock-store'

import { Provider } from 'react-redux'

const mockStore = configureStore();
const initialState = {
    userInfo: {             
        followers: null,
        repositories: null,
        error: false   
    }
}
let store;


beforeEach(()=> {
    store = mockStore(initialState);
})

describe('<AnotherPage />', () => {    
    describe('render()', () => {
        test('renders the component', () => {
            const wrapper = mount(<Provider store={store}><AnotherPage/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
   
        test('renders the component with full data', () => {
            store = mockStore({ 
                userInfo: {             
                    followers: [{
                        id: "id1",
                        login: "login1"
                    }],
                    repositories: [{
                        id: "id1",
                        name: "name1"
                    }],
                    error: false   
                }
            })
            const wrapper = mount(<Provider store={store}><AnotherPage/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
        
        test('renders the component with empty data', () => {
            store = mockStore({ 
                userInfo: {             
                    followers: [],
                    repositories: [],
                    error: true   
                }
            })
            const wrapper = mount(<Provider store={store}><AnotherPage/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot()
        })
      })
})

