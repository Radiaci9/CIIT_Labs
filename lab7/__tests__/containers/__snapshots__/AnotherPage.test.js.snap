// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnotherPage /> render() renders the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AnotherPage)>
    <AnotherPage
      fetchUserFollowers={[Function]}
      fetchUserRepositories={[Function]}
      user={
        Object {
          "error": false,
          "followers": null,
          "repositories": null,
        }
      }
    >
      <Row
        className="col-sm-11"
        style={
          Object {
            "display": "flex",
            "margin": "20px auto 20px auto",
          }
        }
        tag="div"
      >
        <div
          className="col-sm-11 row"
          style={
            Object {
              "display": "flex",
              "margin": "20px auto 20px auto",
            }
          }
        >
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Followers:
              </h2>
              <img
                alt="loading"
                src="../images/Ajax-loader.gif"
                style={
                  Object {
                    "display": "block",
                    "margin": "0 auto",
                  }
                }
              />
            </div>
          </Col>
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Repositories:
              </h2>
              <img
                alt="loading"
                src="../images/Ajax-loader.gif"
                style={
                  Object {
                    "display": "block",
                    "margin": "0 auto",
                  }
                }
              />
            </div>
          </Col>
        </div>
      </Row>
    </AnotherPage>
  </Connect(AnotherPage)>
</Provider>
`;

exports[`<AnotherPage /> render() renders the component with empty data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AnotherPage)>
    <AnotherPage
      fetchUserFollowers={[Function]}
      fetchUserRepositories={[Function]}
      user={
        Object {
          "error": true,
          "followers": Array [],
          "repositories": Array [],
        }
      }
    >
      <Row
        className="col-sm-11"
        style={
          Object {
            "display": "flex",
            "margin": "20px auto 20px auto",
          }
        }
        tag="div"
      >
        <div
          className="col-sm-11 row"
          style={
            Object {
              "display": "flex",
              "margin": "20px auto 20px auto",
            }
          }
        >
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Followers:
              </h2>
              <h4>
                Followers Not Found
              </h4>
            </div>
          </Col>
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Repositories:
              </h2>
              <h4>
                Repositories Not Found
              </h4>
            </div>
          </Col>
        </div>
      </Row>
    </AnotherPage>
  </Connect(AnotherPage)>
</Provider>
`;

exports[`<AnotherPage /> render() renders the component with full data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AnotherPage)>
    <AnotherPage
      fetchUserFollowers={[Function]}
      fetchUserRepositories={[Function]}
      user={
        Object {
          "error": false,
          "followers": Array [
            Object {
              "id": "id1",
              "login": "login1",
            },
          ],
          "repositories": Array [
            Object {
              "id": "id1",
              "name": "name1",
            },
          ],
        }
      }
    >
      <Row
        className="col-sm-11"
        style={
          Object {
            "display": "flex",
            "margin": "20px auto 20px auto",
          }
        }
        tag="div"
      >
        <div
          className="col-sm-11 row"
          style={
            Object {
              "display": "flex",
              "margin": "20px auto 20px auto",
            }
          }
        >
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Followers:
              </h2>
              <ul
                className="list=group mb-3"
                key="id1"
              >
                <li
                  className="list-group-item"
                  key="Idid1"
                >
                  Id
                  : 
                  id1
                </li>
                <li
                  className="list-group-item"
                  key="Loginlogin1"
                >
                  Login
                  : 
                  login1
                </li>
              </ul>
            </div>
          </Col>
          <Col
            sm={
              Object {
                "size": 6,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Repositories:
              </h2>
              <ul
                className="list=group mb-3"
                key="id1"
              >
                <li
                  className="list-group-item"
                  key="Idid1"
                >
                  Id
                  : 
                  id1
                </li>
                <li
                  className="list-group-item"
                  key="Namename1"
                >
                  Name
                  : 
                  name1
                </li>
              </ul>
            </div>
          </Col>
        </div>
      </Row>
    </AnotherPage>
  </Connect(AnotherPage)>
</Provider>
`;
